{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Title"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Part 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Dependencies\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats as sp\n",
    "from Bio.KEGG import REST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "hsa_gene_dict = pd.read_pickle('hsa_gene_dict.p')\n",
    "Metabolomics_data = pd.read_pickle('Metabolomics_data.p')\n",
    "swath_lod = pd.read_pickle('SWATH_LoD.p')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "cpd_pw_text = REST.kegg_link(\"pathway\",\"cpd\").read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'cpd:C00022\\tpath:map00010\\ncpd:C00024\\tpath:map00010\\ncpd:C00031\\tpath:map00010\\ncpd:C00033\\tpath:map00010\\n'"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cpd_pw_text[0:100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "cpd_pw_text = cpd_pw_text.split('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['cpd:C00022\\tpath:map00010',\n",
       " 'cpd:C00024\\tpath:map00010',\n",
       " 'cpd:C00031\\tpath:map00010',\n",
       " 'cpd:C00033\\tpath:map00010',\n",
       " 'cpd:C00036\\tpath:map00010',\n",
       " 'cpd:C00068\\tpath:map00010',\n",
       " 'cpd:C00074\\tpath:map00010',\n",
       " 'cpd:C00084\\tpath:map00010',\n",
       " 'cpd:C00103\\tpath:map00010',\n",
       " 'cpd:C00111\\tpath:map00010']"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cpd_pw_text[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathway_cpd = []\n",
    "for item in cpd_pw_text:\n",
    "    if '\\t' in item:\n",
    "        item_split = item.split('\\t')\n",
    "        cpd_string = item_split[0]\n",
    "        cpd_string = cpd_string[4:]\n",
    "        \n",
    "        map_string = item_split[1]\n",
    "        map_string = map_string[5:]\n",
    "        \n",
    "        curr_list = [cpd_string,map_string]\n",
    "        pathway_cpd.append(curr_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['C00022', 'map00010'],\n",
       " ['C00024', 'map00010'],\n",
       " ['C00031', 'map00010'],\n",
       " ['C00033', 'map00010'],\n",
       " ['C00036', 'map00010'],\n",
       " ['C00068', 'map00010'],\n",
       " ['C00074', 'map00010'],\n",
       " ['C00084', 'map00010'],\n",
       " ['C00103', 'map00010'],\n",
       " ['C00111', 'map00010']]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathway_cpd[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010',\n",
       " 'map00010']"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "map_list = [curr_list[1] for curr_list in pathway_cpd]\n",
    "map_list_unique= list(set(map_list))\n",
    "map_list[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['map00010', 'Glycolysis / Gluconeogenesis'],\n",
       " ['map00020', 'Citrate cycle (TCA cycle)'],\n",
       " ['map00030', 'Pentose phosphate pathway'],\n",
       " ['map00040', 'Pentose and glucuronate interconversions'],\n",
       " ['map00051', 'Fructose and mannose metabolism'],\n",
       " ['map00052', 'Galactose metabolism'],\n",
       " ['map00053', 'Ascorbate and aldarate metabolism'],\n",
       " ['map00061', 'Fatty acid biosynthesis'],\n",
       " ['map00062', 'Fatty acid elongation'],\n",
       " ['map00071', 'Fatty acid degradation'],\n",
       " ['map00072', 'Synthesis and degradation of ketone bodies'],\n",
       " ['map00073', 'Cutin, suberine and wax biosynthesis'],\n",
       " ['map00100', 'Steroid biosynthesis'],\n",
       " ['map00120', 'Primary bile acid biosynthesis'],\n",
       " ['map00121', 'Secondary bile acid biosynthesis'],\n",
       " ['map00130', 'Ubiquinone and other terpenoid-quinone biosynthesis'],\n",
       " ['map00140', 'Steroid hormone biosynthesis'],\n",
       " ['map00190', 'Oxidative phosphorylation'],\n",
       " ['map00195', 'Photosynthesis'],\n",
       " ['map00196', 'Photosynthesis - antenna proteins'],\n",
       " ['map00220', 'Arginine biosynthesis'],\n",
       " ['map00230', 'Purine metabolism'],\n",
       " ['map00232', 'Caffeine metabolism'],\n",
       " ['map00240', 'Pyrimidine metabolism'],\n",
       " ['map00250', 'Alanine, aspartate and glutamate metabolism'],\n",
       " ['map00253', 'Tetracycline biosynthesis'],\n",
       " ['map00254', 'Aflatoxin biosynthesis'],\n",
       " ['map00260', 'Glycine, serine and threonine metabolism'],\n",
       " ['map00261', 'Monobactam biosynthesis'],\n",
       " ['map00270', 'Cysteine and methionine metabolism'],\n",
       " ['map00280', 'Valine, leucine and isoleucine degradation'],\n",
       " ['map00281', 'Geraniol degradation'],\n",
       " ['map00290', 'Valine, leucine and isoleucine biosynthesis'],\n",
       " ['map00300', 'Lysine biosynthesis'],\n",
       " ['map00310', 'Lysine degradation'],\n",
       " ['map00311', 'Penicillin and cephalosporin biosynthesis'],\n",
       " ['map00330', 'Arginine and proline metabolism'],\n",
       " ['map00331', 'Clavulanic acid biosynthesis'],\n",
       " ['map00332', 'Carbapenem biosynthesis'],\n",
       " ['map00333', 'Prodigiosin biosynthesis'],\n",
       " ['map00340', 'Histidine metabolism'],\n",
       " ['map00350', 'Tyrosine metabolism'],\n",
       " ['map00360', 'Phenylalanine metabolism'],\n",
       " ['map00361', 'Chlorocyclohexane and chlorobenzene degradation'],\n",
       " ['map00362', 'Benzoate degradation'],\n",
       " ['map00363', 'Bisphenol degradation'],\n",
       " ['map00364', 'Fluorobenzoate degradation'],\n",
       " ['map00365', 'Furfural degradation'],\n",
       " ['map00380', 'Tryptophan metabolism'],\n",
       " ['map00400', 'Phenylalanine, tyrosine and tryptophan biosynthesis'],\n",
       " ['map00401', 'Novobiocin biosynthesis'],\n",
       " ['map00402', 'Benzoxazinoid biosynthesis'],\n",
       " ['map00403', 'Indole diterpene alkaloid biosynthesis'],\n",
       " ['map00404', 'Staurosporine biosynthesis'],\n",
       " ['map00405', 'Phenazine biosynthesis'],\n",
       " ['map00410', 'beta-Alanine metabolism'],\n",
       " ['map00430', 'Taurine and hypotaurine metabolism'],\n",
       " ['map00440', 'Phosphonate and phosphinate metabolism'],\n",
       " ['map00450', 'Selenocompound metabolism'],\n",
       " ['map00460', 'Cyanoamino acid metabolism'],\n",
       " ['map00471', 'D-Glutamine and D-glutamate metabolism'],\n",
       " ['map00472', 'D-Arginine and D-ornithine metabolism'],\n",
       " ['map00473', 'D-Alanine metabolism'],\n",
       " ['map00480', 'Glutathione metabolism'],\n",
       " ['map00500', 'Starch and sucrose metabolism'],\n",
       " ['map00510', 'N-Glycan biosynthesis'],\n",
       " ['map00511', 'Other glycan degradation'],\n",
       " ['map00512', 'Mucin type O-glycan biosynthesis'],\n",
       " ['map00513', 'Various types of N-glycan biosynthesis'],\n",
       " ['map00514', 'Other types of O-glycan biosynthesis'],\n",
       " ['map00515', 'Mannose type O-glycan biosynthesis'],\n",
       " ['map00520', 'Amino sugar and nucleotide sugar metabolism'],\n",
       " ['map00521', 'Streptomycin biosynthesis'],\n",
       " ['map00522', 'Biosynthesis of 12-, 14- and 16-membered macrolides'],\n",
       " ['map00523', 'Polyketide sugar unit biosynthesis'],\n",
       " ['map00524', 'Neomycin, kanamycin and gentamicin biosynthesis'],\n",
       " ['map00525', 'Acarbose and validamycin biosynthesis'],\n",
       " ['map00531', 'Glycosaminoglycan degradation'],\n",
       " ['map00532',\n",
       "  'Glycosaminoglycan biosynthesis - chondroitin sulfate / dermatan sulfate'],\n",
       " ['map00533', 'Glycosaminoglycan biosynthesis - keratan sulfate'],\n",
       " ['map00534', 'Glycosaminoglycan biosynthesis - heparan sulfate / heparin'],\n",
       " ['map00540', 'Lipopolysaccharide biosynthesis'],\n",
       " ['map00550', 'Peptidoglycan biosynthesis'],\n",
       " ['map00561', 'Glycerolipid metabolism'],\n",
       " ['map00562', 'Inositol phosphate metabolism'],\n",
       " ['map00563', 'Glycosylphosphatidylinositol (GPI)-anchor biosynthesis'],\n",
       " ['map00564', 'Glycerophospholipid metabolism'],\n",
       " ['map00565', 'Ether lipid metabolism'],\n",
       " ['map00571', 'Lipoarabinomannan (LAM) biosynthesis'],\n",
       " ['map00572', 'Arabinogalactan biosynthesis - Mycobacterium'],\n",
       " ['map00590', 'Arachidonic acid metabolism'],\n",
       " ['map00591', 'Linoleic acid metabolism'],\n",
       " ['map00592', 'alpha-Linolenic acid metabolism'],\n",
       " ['map00600', 'Sphingolipid metabolism'],\n",
       " ['map00601', 'Glycosphingolipid biosynthesis - lacto and neolacto series'],\n",
       " ['map00603', 'Glycosphingolipid biosynthesis - globo and isoglobo series'],\n",
       " ['map00604', 'Glycosphingolipid biosynthesis - ganglio series'],\n",
       " ['map00620', 'Pyruvate metabolism'],\n",
       " ['map00621', 'Dioxin degradation'],\n",
       " ['map00622', 'Xylene degradation'],\n",
       " ['map00623', 'Toluene degradation'],\n",
       " ['map00624', 'Polycyclic aromatic hydrocarbon degradation'],\n",
       " ['map00625', 'Chloroalkane and chloroalkene degradation'],\n",
       " ['map00626', 'Naphthalene degradation'],\n",
       " ['map00627', 'Aminobenzoate degradation'],\n",
       " ['map00630', 'Glyoxylate and dicarboxylate metabolism'],\n",
       " ['map00633', 'Nitrotoluene degradation'],\n",
       " ['map00640', 'Propanoate metabolism'],\n",
       " ['map00642', 'Ethylbenzene degradation'],\n",
       " ['map00643', 'Styrene degradation'],\n",
       " ['map00650', 'Butanoate metabolism'],\n",
       " ['map00660', 'C5-Branched dibasic acid metabolism'],\n",
       " ['map00670', 'One carbon pool by folate'],\n",
       " ['map00680', 'Methane metabolism'],\n",
       " ['map00710', 'Carbon fixation in photosynthetic organisms'],\n",
       " ['map00720', 'Carbon fixation pathways in prokaryotes'],\n",
       " ['map00730', 'Thiamine metabolism'],\n",
       " ['map00740', 'Riboflavin metabolism'],\n",
       " ['map00750', 'Vitamin B6 metabolism'],\n",
       " ['map00760', 'Nicotinate and nicotinamide metabolism'],\n",
       " ['map00770', 'Pantothenate and CoA biosynthesis'],\n",
       " ['map00780', 'Biotin metabolism'],\n",
       " ['map00785', 'Lipoic acid metabolism'],\n",
       " ['map00790', 'Folate biosynthesis'],\n",
       " ['map00791', 'Atrazine degradation'],\n",
       " ['map00830', 'Retinol metabolism'],\n",
       " ['map00860', 'Porphyrin and chlorophyll metabolism'],\n",
       " ['map00900', 'Terpenoid backbone biosynthesis'],\n",
       " ['map00901', 'Indole alkaloid biosynthesis'],\n",
       " ['map00902', 'Monoterpenoid biosynthesis'],\n",
       " ['map00903', 'Limonene and pinene degradation'],\n",
       " ['map00904', 'Diterpenoid biosynthesis'],\n",
       " ['map00905', 'Brassinosteroid biosynthesis'],\n",
       " ['map00906', 'Carotenoid biosynthesis'],\n",
       " ['map00908', 'Zeatin biosynthesis'],\n",
       " ['map00909', 'Sesquiterpenoid and triterpenoid biosynthesis'],\n",
       " ['map00910', 'Nitrogen metabolism'],\n",
       " ['map00920', 'Sulfur metabolism'],\n",
       " ['map00930', 'Caprolactam degradation'],\n",
       " ['map00940', 'Phenylpropanoid biosynthesis'],\n",
       " ['map00941', 'Flavonoid biosynthesis'],\n",
       " ['map00942', 'Anthocyanin biosynthesis'],\n",
       " ['map00943', 'Isoflavonoid biosynthesis'],\n",
       " ['map00944', 'Flavone and flavonol biosynthesis'],\n",
       " ['map00945', 'Stilbenoid, diarylheptanoid and gingerol biosynthesis'],\n",
       " ['map00950', 'Isoquinoline alkaloid biosynthesis'],\n",
       " ['map00960', 'Tropane, piperidine and pyridine alkaloid biosynthesis'],\n",
       " ['map00965', 'Betalain biosynthesis'],\n",
       " ['map00966', 'Glucosinolate biosynthesis'],\n",
       " ['map00970', 'Aminoacyl-tRNA biosynthesis'],\n",
       " ['map00980', 'Metabolism of xenobiotics by cytochrome P450'],\n",
       " ['map00981', 'Insect hormone biosynthesis'],\n",
       " ['map00982', 'Drug metabolism - cytochrome P450'],\n",
       " ['map00983', 'Drug metabolism - other enzymes'],\n",
       " ['map00984', 'Steroid degradation'],\n",
       " ['map00997', 'Biosynthesis of various secondary metabolites - part 3'],\n",
       " ['map00998', 'Biosynthesis of various secondary metabolites - part 2'],\n",
       " ['map00999', 'Biosynthesis of various secondary metabolites - part 1'],\n",
       " ['map01010', 'Overview of biosynthetic pathways'],\n",
       " ['map01040', 'Biosynthesis of unsaturated fatty acids'],\n",
       " ['map01051', 'Biosynthesis of ansamycins'],\n",
       " ['map01052', 'Type I polyketide structures'],\n",
       " ['map01053', 'Biosynthesis of siderophore group nonribosomal peptides'],\n",
       " ['map01054', 'Nonribosomal peptide structures'],\n",
       " ['map01055', 'Biosynthesis of vancomycin group antibiotics'],\n",
       " ['map01056', 'Biosynthesis of type II polyketide backbone'],\n",
       " ['map01057', 'Biosynthesis of type II polyketide products'],\n",
       " ['map01058', 'Acridone alkaloid biosynthesis'],\n",
       " ['map01059', 'Biosynthesis of enediyne antibiotics'],\n",
       " ['map01060', 'Biosynthesis of plant secondary metabolites'],\n",
       " ['map01061', 'Biosynthesis of phenylpropanoids'],\n",
       " ['map01062', 'Biosynthesis of terpenoids and steroids'],\n",
       " ['map01063', 'Biosynthesis of alkaloids derived from shikimate pathway'],\n",
       " ['map01064',\n",
       "  'Biosynthesis of alkaloids derived from ornithine, lysine and nicotinic acid'],\n",
       " ['map01065', 'Biosynthesis of alkaloids derived from histidine and purine'],\n",
       " ['map01066',\n",
       "  'Biosynthesis of alkaloids derived from terpenoid and polyketide'],\n",
       " ['map01070', 'Biosynthesis of plant hormones'],\n",
       " ['map01100', 'Metabolic pathways'],\n",
       " ['map01110', 'Biosynthesis of secondary metabolites'],\n",
       " ['map01120', 'Microbial metabolism in diverse environments'],\n",
       " ['map01200', 'Carbon metabolism'],\n",
       " ['map01210', '2-Oxocarboxylic acid metabolism'],\n",
       " ['map01212', 'Fatty acid metabolism'],\n",
       " ['map01220', 'Degradation of aromatic compounds'],\n",
       " ['map01230', 'Biosynthesis of amino acids'],\n",
       " ['map01501', 'beta-Lactam resistance'],\n",
       " ['map01502', 'Vancomycin resistance'],\n",
       " ['map01503', 'Cationic antimicrobial peptide (CAMP) resistance'],\n",
       " ['map01521', 'EGFR tyrosine kinase inhibitor resistance'],\n",
       " ['map01522', 'Endocrine resistance'],\n",
       " ['map01523', 'Antifolate resistance'],\n",
       " ['map01524', 'Platinum drug resistance'],\n",
       " ['map02010', 'ABC transporters'],\n",
       " ['map02020', 'Two-component system'],\n",
       " ['map02024', 'Quorum sensing'],\n",
       " ['map02025', 'Biofilm formation - Pseudomonas aeruginosa'],\n",
       " ['map02026', 'Biofilm formation - Escherichia coli'],\n",
       " ['map02030', 'Bacterial chemotaxis'],\n",
       " ['map02040', 'Flagellar assembly'],\n",
       " ['map02060', 'Phosphotransferase system (PTS)'],\n",
       " ['map03008', 'Ribosome biogenesis in eukaryotes'],\n",
       " ['map03010', 'Ribosome'],\n",
       " ['map03013', 'RNA transport'],\n",
       " ['map03015', 'mRNA surveillance pathway'],\n",
       " ['map03018', 'RNA degradation'],\n",
       " ['map03020', 'RNA polymerase'],\n",
       " ['map03022', 'Basal transcription factors'],\n",
       " ['map03030', 'DNA replication'],\n",
       " ['map03040', 'Spliceosome'],\n",
       " ['map03050', 'Proteasome'],\n",
       " ['map03060', 'Protein export'],\n",
       " ['map03070', 'Bacterial secretion system'],\n",
       " ['map03320', 'PPAR signaling pathway'],\n",
       " ['map03410', 'Base excision repair'],\n",
       " ['map03420', 'Nucleotide excision repair'],\n",
       " ['map03430', 'Mismatch repair'],\n",
       " ['map03440', 'Homologous recombination'],\n",
       " ['map03450', 'Non-homologous end-joining'],\n",
       " ['map03460', 'Fanconi anemia pathway'],\n",
       " ['map04010', 'MAPK signaling pathway'],\n",
       " ['map04011', 'MAPK signaling pathway - yeast'],\n",
       " ['map04012', 'ErbB signaling pathway'],\n",
       " ['map04013', 'MAPK signaling pathway - fly'],\n",
       " ['map04014', 'Ras signaling pathway'],\n",
       " ['map04015', 'Rap1 signaling pathway'],\n",
       " ['map04016', 'MAPK signaling pathway - plant'],\n",
       " ['map04020', 'Calcium signaling pathway'],\n",
       " ['map04022', 'cGMP-PKG signaling pathway'],\n",
       " ['map04024', 'cAMP signaling pathway'],\n",
       " ['map04060', 'Cytokine-cytokine receptor interaction'],\n",
       " ['map04061', 'Viral protein interaction with cytokine and cytokine receptor'],\n",
       " ['map04062', 'Chemokine signaling pathway'],\n",
       " ['map04064', 'NF-kappa B signaling pathway'],\n",
       " ['map04066', 'HIF-1 signaling pathway'],\n",
       " ['map04068', 'FoxO signaling pathway'],\n",
       " ['map04070', 'Phosphatidylinositol signaling system'],\n",
       " ['map04071', 'Sphingolipid signaling pathway'],\n",
       " ['map04072', 'Phospholipase D signaling pathway'],\n",
       " ['map04075', 'Plant hormone signal transduction'],\n",
       " ['map04080', 'Neuroactive ligand-receptor interaction'],\n",
       " ['map04110', 'Cell cycle'],\n",
       " ['map04111', 'Cell cycle - yeast'],\n",
       " ['map04112', 'Cell cycle - Caulobacter'],\n",
       " ['map04113', 'Meiosis - yeast'],\n",
       " ['map04114', 'Oocyte meiosis'],\n",
       " ['map04115', 'p53 signaling pathway'],\n",
       " ['map04120', 'Ubiquitin mediated proteolysis'],\n",
       " ['map04122', 'Sulfur relay system'],\n",
       " ['map04130', 'SNARE interactions in vesicular transport'],\n",
       " ['map04136', 'Autophagy - other'],\n",
       " ['map04137', 'Mitophagy - animal'],\n",
       " ['map04138', 'Autophagy - yeast'],\n",
       " ['map04139', 'Mitophagy - yeast'],\n",
       " ['map04140', 'Autophagy - animal'],\n",
       " ['map04141', 'Protein processing in endoplasmic reticulum'],\n",
       " ['map04142', 'Lysosome'],\n",
       " ['map04144', 'Endocytosis'],\n",
       " ['map04145', 'Phagosome'],\n",
       " ['map04146', 'Peroxisome'],\n",
       " ['map04150', 'mTOR signaling pathway'],\n",
       " ['map04151', 'PI3K-Akt signaling pathway'],\n",
       " ['map04152', 'AMPK signaling pathway'],\n",
       " ['map04210', 'Apoptosis'],\n",
       " ['map04211', 'Longevity regulating pathway'],\n",
       " ['map04212', 'Longevity regulating pathway - worm'],\n",
       " ['map04213', 'Longevity regulating pathway - multiple species'],\n",
       " ['map04214', 'Apoptosis - fly'],\n",
       " ['map04215', 'Apoptosis - multiple species'],\n",
       " ['map04216', 'Ferroptosis'],\n",
       " ['map04217', 'Necroptosis'],\n",
       " ['map04218', 'Cellular senescence'],\n",
       " ['map04260', 'Cardiac muscle contraction'],\n",
       " ['map04261', 'Adrenergic signaling in cardiomyocytes'],\n",
       " ['map04270', 'Vascular smooth muscle contraction'],\n",
       " ['map04310', 'Wnt signaling pathway'],\n",
       " ['map04320', 'Dorso-ventral axis formation'],\n",
       " ['map04330', 'Notch signaling pathway'],\n",
       " ['map04340', 'Hedgehog signaling pathway'],\n",
       " ['map04341', 'Hedgehog signaling pathway - fly'],\n",
       " ['map04350', 'TGF-beta signaling pathway'],\n",
       " ['map04360', 'Axon guidance'],\n",
       " ['map04361', 'Axon regeneration'],\n",
       " ['map04370', 'VEGF signaling pathway'],\n",
       " ['map04371', 'Apelin signaling pathway'],\n",
       " ['map04380', 'Osteoclast differentiation'],\n",
       " ['map04390', 'Hippo signaling pathway'],\n",
       " ['map04391', 'Hippo signaling pathway - fly'],\n",
       " ['map04392', 'Hippo signaling pathway - multiple species'],\n",
       " ['map04510', 'Focal adhesion'],\n",
       " ['map04512', 'ECM-receptor interaction'],\n",
       " ['map04514', 'Cell adhesion molecules (CAMs)'],\n",
       " ['map04520', 'Adherens junction'],\n",
       " ['map04530', 'Tight junction'],\n",
       " ['map04540', 'Gap junction'],\n",
       " ['map04550', 'Signaling pathways regulating pluripotency of stem cells'],\n",
       " ['map04610', 'Complement and coagulation cascades'],\n",
       " ['map04611', 'Platelet activation'],\n",
       " ['map04612', 'Antigen processing and presentation'],\n",
       " ['map04614', 'Renin-angiotensin system'],\n",
       " ['map04620', 'Toll-like receptor signaling pathway'],\n",
       " ['map04621', 'NOD-like receptor signaling pathway'],\n",
       " ['map04622', 'RIG-I-like receptor signaling pathway'],\n",
       " ['map04623', 'Cytosolic DNA-sensing pathway'],\n",
       " ['map04624', 'Toll and Imd signaling pathway'],\n",
       " ['map04625', 'C-type lectin receptor signaling pathway'],\n",
       " ['map04626', 'Plant-pathogen interaction'],\n",
       " ['map04630', 'JAK-STAT signaling pathway'],\n",
       " ['map04640', 'Hematopoietic cell lineage'],\n",
       " ['map04650', 'Natural killer cell mediated cytotoxicity'],\n",
       " ['map04657', 'IL-17 signaling pathway'],\n",
       " ['map04658', 'Th1 and Th2 cell differentiation'],\n",
       " ['map04659', 'Th17 cell differentiation'],\n",
       " ['map04660', 'T cell receptor signaling pathway'],\n",
       " ['map04662', 'B cell receptor signaling pathway'],\n",
       " ['map04664', 'Fc epsilon RI signaling pathway'],\n",
       " ['map04666', 'Fc gamma R-mediated phagocytosis'],\n",
       " ['map04668', 'TNF signaling pathway'],\n",
       " ['map04670', 'Leukocyte transendothelial migration'],\n",
       " ['map04672', 'Intestinal immune network for IgA production'],\n",
       " ['map04710', 'Circadian rhythm'],\n",
       " ['map04711', 'Circadian rhythm - fly'],\n",
       " ['map04712', 'Circadian rhythm - plant'],\n",
       " ['map04713', 'Circadian entrainment'],\n",
       " ['map04714', 'Thermogenesis'],\n",
       " ['map04720', 'Long-term potentiation'],\n",
       " ['map04721', 'Synaptic vesicle cycle'],\n",
       " ['map04722', 'Neurotrophin signaling pathway'],\n",
       " ['map04723', 'Retrograde endocannabinoid signaling'],\n",
       " ['map04724', 'Glutamatergic synapse'],\n",
       " ['map04725', 'Cholinergic synapse'],\n",
       " ['map04726', 'Serotonergic synapse'],\n",
       " ['map04727', 'GABAergic synapse'],\n",
       " ['map04728', 'Dopaminergic synapse'],\n",
       " ['map04730', 'Long-term depression'],\n",
       " ['map04740', 'Olfactory transduction'],\n",
       " ['map04742', 'Taste transduction'],\n",
       " ['map04744', 'Phototransduction'],\n",
       " ['map04745', 'Phototransduction - fly'],\n",
       " ['map04750', 'Inflammatory mediator regulation of TRP channels'],\n",
       " ['map04810', 'Regulation of actin cytoskeleton'],\n",
       " ['map04910', 'Insulin signaling pathway'],\n",
       " ['map04911', 'Insulin secretion'],\n",
       " ['map04912', 'GnRH signaling pathway'],\n",
       " ['map04913', 'Ovarian steroidogenesis'],\n",
       " ['map04914', 'Progesterone-mediated oocyte maturation'],\n",
       " ['map04915', 'Estrogen signaling pathway'],\n",
       " ['map04916', 'Melanogenesis'],\n",
       " ['map04917', 'Prolactin signaling pathway'],\n",
       " ['map04918', 'Thyroid hormone synthesis'],\n",
       " ['map04919', 'Thyroid hormone signaling pathway'],\n",
       " ['map04920', 'Adipocytokine signaling pathway'],\n",
       " ['map04921', 'Oxytocin signaling pathway'],\n",
       " ['map04922', 'Glucagon signaling pathway'],\n",
       " ['map04923', 'Regulation of lipolysis in adipocytes'],\n",
       " ['map04924', 'Renin secretion'],\n",
       " ['map04925', 'Aldosterone synthesis and secretion'],\n",
       " ['map04926', 'Relaxin signaling pathway'],\n",
       " ['map04927', 'Cortisol synthesis and secretion'],\n",
       " ['map04928', 'Parathyroid hormone synthesis, secretion and action'],\n",
       " ['map04929', 'GnRH secretion'],\n",
       " ['map04930', 'Type II diabetes mellitus'],\n",
       " ['map04931', 'Insulin resistance'],\n",
       " ['map04932', 'Non-alcoholic fatty liver disease (NAFLD)'],\n",
       " ['map04933', 'AGE-RAGE signaling pathway in diabetic complications'],\n",
       " ['map04934', 'Cushing syndrome'],\n",
       " ['map04935', 'Growth hormone synthesis, secretion and action'],\n",
       " ['map04940', 'Type I diabetes mellitus'],\n",
       " ['map04950', 'Maturity onset diabetes of the young'],\n",
       " ['map04960', 'Aldosterone-regulated sodium reabsorption'],\n",
       " ['map04961', 'Endocrine and other factor-regulated calcium reabsorption'],\n",
       " ['map04962', 'Vasopressin-regulated water reabsorption'],\n",
       " ['map04964', 'Proximal tubule bicarbonate reclamation'],\n",
       " ['map04966', 'Collecting duct acid secretion'],\n",
       " ['map04970', 'Salivary secretion'],\n",
       " ['map04971', 'Gastric acid secretion'],\n",
       " ['map04972', 'Pancreatic secretion'],\n",
       " ['map04973', 'Carbohydrate digestion and absorption'],\n",
       " ['map04974', 'Protein digestion and absorption'],\n",
       " ['map04975', 'Fat digestion and absorption'],\n",
       " ['map04976', 'Bile secretion'],\n",
       " ['map04977', 'Vitamin digestion and absorption'],\n",
       " ['map04978', 'Mineral absorption'],\n",
       " ['map04979', 'Cholesterol metabolism'],\n",
       " ['map05010', 'Alzheimer disease'],\n",
       " ['map05012', 'Parkinson disease'],\n",
       " ['map05014', 'Amyotrophic lateral sclerosis (ALS)'],\n",
       " ['map05016', 'Huntington disease'],\n",
       " ['map05017', 'Spinocerebellar ataxia'],\n",
       " ['map05020', 'Prion diseases'],\n",
       " ['map05030', 'Cocaine addiction'],\n",
       " ['map05031', 'Amphetamine addiction'],\n",
       " ['map05032', 'Morphine addiction'],\n",
       " ['map05033', 'Nicotine addiction'],\n",
       " ['map05034', 'Alcoholism'],\n",
       " ['map05100', 'Bacterial invasion of epithelial cells'],\n",
       " ['map05110', 'Vibrio cholerae infection'],\n",
       " ['map05111', 'Biofilm formation - Vibrio cholerae'],\n",
       " ['map05120', 'Epithelial cell signaling in Helicobacter pylori infection'],\n",
       " ['map05130', 'Pathogenic Escherichia coli infection'],\n",
       " ['map05131', 'Shigellosis'],\n",
       " ['map05132', 'Salmonella infection'],\n",
       " ['map05133', 'Pertussis'],\n",
       " ['map05134', 'Legionellosis'],\n",
       " ['map05135', 'Yersinia infection'],\n",
       " ['map05140', 'Leishmaniasis'],\n",
       " ['map05142', 'Chagas disease (American trypanosomiasis)'],\n",
       " ['map05143', 'African trypanosomiasis'],\n",
       " ['map05144', 'Malaria'],\n",
       " ['map05145', 'Toxoplasmosis'],\n",
       " ['map05146', 'Amoebiasis'],\n",
       " ['map05150', 'Staphylococcus aureus infection'],\n",
       " ['map05152', 'Tuberculosis'],\n",
       " ['map05160', 'Hepatitis C'],\n",
       " ['map05161', 'Hepatitis B'],\n",
       " ['map05162', 'Measles'],\n",
       " ['map05163', 'Human cytomegalovirus infection'],\n",
       " ['map05164', 'Influenza A'],\n",
       " ['map05165', 'Human papillomavirus infection'],\n",
       " ['map05166', 'Human T-cell leukemia virus 1 infection'],\n",
       " ['map05167', 'Kaposi sarcoma-associated herpesvirus infection'],\n",
       " ['map05168', 'Herpes simplex virus 1 infection'],\n",
       " ['map05169', 'Epstein-Barr virus infection'],\n",
       " ['map05170', 'Human immunodeficiency virus 1 infection'],\n",
       " ['map05200', 'Pathways in cancer'],\n",
       " ['map05202', 'Transcriptional misregulation in cancer'],\n",
       " ['map05203', 'Viral carcinogenesis'],\n",
       " ['map05204', 'Chemical carcinogenesis'],\n",
       " ['map05205', 'Proteoglycans in cancer'],\n",
       " ['map05206', 'MicroRNAs in cancer'],\n",
       " ['map05210', 'Colorectal cancer'],\n",
       " ['map05211', 'Renal cell carcinoma'],\n",
       " ['map05212', 'Pancreatic cancer'],\n",
       " ['map05213', 'Endometrial cancer'],\n",
       " ['map05214', 'Glioma'],\n",
       " ['map05215', 'Prostate cancer'],\n",
       " ['map05216', 'Thyroid cancer'],\n",
       " ['map05217', 'Basal cell carcinoma'],\n",
       " ['map05218', 'Melanoma'],\n",
       " ['map05219', 'Bladder cancer'],\n",
       " ['map05220', 'Chronic myeloid leukemia'],\n",
       " ['map05221', 'Acute myeloid leukemia'],\n",
       " ['map05222', 'Small cell lung cancer'],\n",
       " ['map05223', 'Non-small cell lung cancer'],\n",
       " ['map05224', 'Breast cancer'],\n",
       " ['map05225', 'Hepatocellular carcinoma'],\n",
       " ['map05226', 'Gastric cancer'],\n",
       " ['map05230', 'Central carbon metabolism in cancer'],\n",
       " ['map05231', 'Choline metabolism in cancer'],\n",
       " ['map05235', 'PD-L1 expression and PD-1 checkpoint pathway in cancer'],\n",
       " ['map05310', 'Asthma'],\n",
       " ['map05320', 'Autoimmune thyroid disease'],\n",
       " ['map05321', 'Inflammatory bowel disease (IBD)'],\n",
       " ['map05322', 'Systemic lupus erythematosus'],\n",
       " ['map05323', 'Rheumatoid arthritis'],\n",
       " ['map05330', 'Allograft rejection'],\n",
       " ['map05332', 'Graft-versus-host disease'],\n",
       " ['map05340', 'Primary immunodeficiency'],\n",
       " ['map05410', 'Hypertrophic cardiomyopathy (HCM)'],\n",
       " ['map05412', 'Arrhythmogenic right ventricular cardiomyopathy (ARVC)'],\n",
       " ['map05414', 'Dilated cardiomyopathy (DCM)'],\n",
       " ['map05416', 'Viral myocarditis'],\n",
       " ['map05418', 'Fluid shear stress and atherosclerosis'],\n",
       " ['map07011', 'Penicillins'],\n",
       " ['map07012', 'Cephalosporins - parenteral agents'],\n",
       " ['map07013', 'Cephalosporins - oral agents'],\n",
       " ['map07014', 'Quinolones'],\n",
       " ['map07015', 'Local analgesics'],\n",
       " ['map07016', 'Sulfonamide derivatives - sulfa drugs'],\n",
       " ['map07017', 'Sulfonamide derivatives - diuretics'],\n",
       " ['map07018', 'Sulfonamide derivatives - hypoglycemic agents'],\n",
       " ['map07019', 'Tetracyclines'],\n",
       " ['map07020', 'Macrolides and ketolides'],\n",
       " ['map07021', 'Aminoglycosides'],\n",
       " ['map07023', 'Rifamycins'],\n",
       " ['map07024', 'HMG-CoA reductase inhibitors'],\n",
       " ['map07025', 'Quinolines'],\n",
       " ['map07026', 'Antifungal agents'],\n",
       " ['map07027', 'Antidepressants'],\n",
       " ['map07028', 'Antipsychotics'],\n",
       " ['map07029', 'Antipsychotics - phenothiazines'],\n",
       " ['map07030', 'Anxiolytics'],\n",
       " ['map07031', 'Antipsychotics - butyrophenones'],\n",
       " ['map07032', 'Hypnotics'],\n",
       " ['map07033', 'Anticonvulsants'],\n",
       " ['map07034', 'Eicosanoids'],\n",
       " ['map07035', 'Prostaglandins'],\n",
       " ['map07036', 'Calcium channel blocking drugs'],\n",
       " ['map07037', 'Antiarrhythmic drugs'],\n",
       " ['map07038', 'Antiulcer drugs'],\n",
       " ['map07039', 'Opioid analgesics'],\n",
       " ['map07040', 'Antineoplastics - alkylating agents'],\n",
       " ['map07041', 'Antineoplastics - antimetabolic agents'],\n",
       " ['map07042', 'Antineoplastics - agents from natural products'],\n",
       " ['map07043', 'Antineoplastics - hormones'],\n",
       " ['map07044', 'Antiviral agents'],\n",
       " ['map07045', 'Antineoplastics - protein kinases inhibitors'],\n",
       " ['map07046', 'Immunosuppressive agents'],\n",
       " ['map07047', 'Osteoporosis drugs'],\n",
       " ['map07048', 'Antimigraines'],\n",
       " ['map07049', 'Antithrombosis agents'],\n",
       " ['map07050', 'Antirheumatics - DMARDs and biological agents'],\n",
       " ['map07051', 'Antidiabetics'],\n",
       " ['map07052', 'Antidyslipidemic agents'],\n",
       " ['map07053', 'Anti-HIV agents'],\n",
       " ['map07054', 'Antiglaucoma agents'],\n",
       " ['map07055', 'Sulfonamide derivatives - overview'],\n",
       " ['map07056', 'Agents for Alzheimer-type dementia'],\n",
       " ['map07057', 'Antiparkinsonian agents'],\n",
       " ['map07110', 'Benzoic acid family'],\n",
       " ['map07112', '1,2-Diphenyl substitution family'],\n",
       " ['map07114', 'Naphthalene family'],\n",
       " ['map07117', 'Benzodiazepine family'],\n",
       " ['map07211', 'Serotonin receptor agonists/antagonists'],\n",
       " ['map07212', 'Histamine H1 receptor antagonists'],\n",
       " ['map07213', 'Dopamine receptor agonists/antagonists'],\n",
       " ['map07214', 'beta-Adrenergic receptor agonists/antagonists'],\n",
       " ['map07215', 'alpha-Adrenergic receptor agonists/antagonists'],\n",
       " ['map07216', 'Catecholamine transferase inhibitors'],\n",
       " ['map07217', 'Renin-angiotensin system inhibitors'],\n",
       " ['map07218', 'HIV protease inhibitors'],\n",
       " ['map07219', 'Cyclooxygenase inhibitors'],\n",
       " ['map07220', 'Cholinergic and anticholinergic drugs'],\n",
       " ['map07221', 'Nicotinic cholinergic receptor antagonists'],\n",
       " ['map07222', 'Peroxisome proliferator-activated receptor (PPAR) agonists'],\n",
       " ['map07223',\n",
       "  'Retinoic acid receptor (RAR) and retinoid X receptor (RXR) agonists/antagonists'],\n",
       " ['map07224', 'Opioid receptor agonists/antagonists'],\n",
       " ['map07225',\n",
       "  'Glucocorticoid and mineralocorticoid receptor agonists/antagonists'],\n",
       " ['map07226',\n",
       "  'Progesterone, androgen and estrogen receptor agonists/antagonists'],\n",
       " ['map07227', 'Histamine H2/H3 receptor agonists/antagonists'],\n",
       " ['map07228', 'Eicosanoid receptor agonists/antagonists'],\n",
       " ['map07229', 'Angiotensin receptor and endothelin receptor antagonists'],\n",
       " ['map07230', 'GABA-A receptor agonists/antagonists'],\n",
       " ['map07231', 'Sodium channel blocking drugs'],\n",
       " ['map07232', 'Potassium channel blocking and opening drugs'],\n",
       " ['map07233', 'Ion transporter inhibitors'],\n",
       " ['map07234', 'Neurotransmitter transporter inhibitors'],\n",
       " ['map07235', 'N-Metyl-D-aspartic acid receptor antagonists']]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathway_names = REST.kegg_list(\"pathway\").read()\n",
    "pathway_names = pathway_names.split('\\n')\n",
    "pathway_list = []\n",
    "\n",
    "for entry in pathway_names:\n",
    "    entry_split = entry.split('\\t')\n",
    "    if len(entry_split)==2:\n",
    "        map_str = entry_split[0][5:]\n",
    "        \n",
    "        path_str = entry_split[1]\n",
    "        curr_list = [map_str,path_str]\n",
    "        pathway_list.append(curr_list)\n",
    "        \n",
    "pathway_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "433"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(map_list_unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "30\n",
      "31\n",
      "32\n",
      "33\n",
      "34\n",
      "35\n",
      "36\n",
      "37\n",
      "38\n",
      "39\n",
      "40\n",
      "41\n",
      "42\n",
      "43\n",
      "44\n",
      "45\n",
      "46\n",
      "47\n",
      "48\n",
      "49\n",
      "50\n",
      "51\n",
      "52\n",
      "53\n",
      "54\n",
      "55\n",
      "56\n",
      "57\n",
      "58\n",
      "59\n",
      "60\n",
      "61\n",
      "62\n",
      "63\n",
      "64\n",
      "65\n",
      "66\n",
      "67\n",
      "68\n",
      "69\n",
      "70\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "81\n",
      "82\n",
      "83\n",
      "84\n",
      "85\n",
      "86\n",
      "87\n",
      "88\n",
      "89\n",
      "90\n",
      "91\n",
      "92\n",
      "93\n",
      "94\n",
      "95\n",
      "96\n",
      "97\n",
      "98\n",
      "99\n",
      "100\n",
      "101\n",
      "102\n",
      "103\n",
      "104\n",
      "105\n",
      "106\n",
      "107\n",
      "108\n",
      "109\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "116\n",
      "117\n",
      "118\n",
      "119\n",
      "120\n",
      "121\n",
      "122\n",
      "123\n",
      "124\n",
      "125\n",
      "126\n",
      "127\n",
      "128\n",
      "129\n",
      "130\n",
      "131\n",
      "132\n",
      "133\n",
      "134\n",
      "135\n",
      "136\n",
      "137\n",
      "138\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "143\n",
      "144\n",
      "145\n",
      "146\n",
      "147\n",
      "148\n",
      "149\n",
      "150\n",
      "151\n",
      "152\n",
      "153\n",
      "154\n",
      "155\n",
      "156\n",
      "157\n",
      "158\n",
      "159\n",
      "160\n",
      "161\n",
      "162\n",
      "163\n",
      "164\n",
      "165\n",
      "166\n",
      "167\n",
      "168\n",
      "169\n",
      "170\n",
      "171\n",
      "172\n",
      "173\n",
      "174\n",
      "175\n",
      "176\n",
      "177\n",
      "178\n",
      "179\n",
      "180\n",
      "181\n",
      "182\n",
      "183\n",
      "184\n",
      "185\n",
      "186\n",
      "187\n",
      "188\n",
      "189\n",
      "190\n",
      "191\n",
      "192\n",
      "193\n",
      "194\n",
      "195\n",
      "196\n",
      "197\n",
      "198\n",
      "199\n",
      "200\n",
      "201\n",
      "202\n",
      "203\n",
      "204\n",
      "205\n",
      "206\n",
      "207\n",
      "208\n",
      "209\n",
      "210\n",
      "211\n",
      "212\n",
      "213\n",
      "214\n",
      "215\n",
      "216\n",
      "217\n",
      "218\n",
      "219\n",
      "220\n",
      "221\n",
      "222\n",
      "223\n",
      "224\n",
      "225\n",
      "226\n",
      "227\n",
      "228\n",
      "229\n",
      "230\n",
      "231\n",
      "232\n",
      "233\n",
      "234\n",
      "235\n",
      "236\n",
      "237\n",
      "238\n",
      "239\n",
      "240\n",
      "241\n",
      "242\n",
      "243\n",
      "244\n",
      "245\n",
      "246\n",
      "247\n",
      "248\n",
      "249\n",
      "250\n",
      "251\n",
      "252\n",
      "253\n",
      "254\n",
      "255\n",
      "256\n",
      "257\n",
      "258\n",
      "259\n",
      "260\n",
      "261\n",
      "262\n",
      "263\n",
      "264\n",
      "265\n",
      "266\n",
      "267\n",
      "268\n",
      "269\n",
      "270\n",
      "271\n",
      "272\n",
      "273\n",
      "274\n",
      "275\n",
      "276\n",
      "277\n",
      "278\n",
      "279\n",
      "280\n",
      "281\n",
      "282\n",
      "283\n",
      "284\n",
      "285\n",
      "286\n",
      "287\n",
      "288\n",
      "289\n",
      "290\n",
      "291\n",
      "292\n",
      "293\n",
      "294\n",
      "295\n",
      "296\n",
      "297\n",
      "298\n",
      "299\n",
      "300\n",
      "301\n",
      "302\n",
      "303\n",
      "304\n",
      "305\n",
      "306\n",
      "307\n",
      "308\n",
      "309\n",
      "310\n",
      "311\n",
      "312\n",
      "313\n",
      "314\n",
      "315\n",
      "316\n",
      "317\n",
      "318\n",
      "319\n",
      "320\n",
      "321\n",
      "322\n",
      "323\n",
      "324\n",
      "325\n",
      "326\n",
      "327\n",
      "328\n",
      "329\n",
      "330\n",
      "331\n",
      "332\n",
      "333\n",
      "334\n",
      "335\n",
      "336\n",
      "337\n",
      "338\n",
      "339\n",
      "340\n",
      "341\n",
      "342\n",
      "343\n",
      "344\n",
      "345\n",
      "346\n",
      "347\n",
      "348\n",
      "349\n",
      "350\n",
      "351\n",
      "352\n",
      "353\n",
      "354\n",
      "355\n",
      "356\n",
      "357\n",
      "358\n",
      "359\n",
      "360\n",
      "361\n",
      "362\n",
      "363\n",
      "364\n",
      "365\n",
      "366\n",
      "367\n",
      "368\n",
      "369\n",
      "370\n",
      "371\n",
      "372\n",
      "373\n",
      "374\n",
      "375\n",
      "376\n",
      "377\n",
      "378\n",
      "379\n",
      "380\n",
      "381\n",
      "382\n",
      "383\n",
      "384\n",
      "385\n",
      "386\n",
      "387\n",
      "388\n",
      "389\n",
      "390\n",
      "391\n",
      "392\n",
      "393\n",
      "394\n",
      "395\n",
      "396\n",
      "397\n",
      "398\n",
      "399\n",
      "400\n",
      "401\n",
      "402\n",
      "403\n",
      "404\n",
      "405\n",
      "406\n",
      "407\n",
      "408\n",
      "409\n",
      "410\n",
      "411\n",
      "412\n",
      "413\n",
      "414\n",
      "415\n",
      "416\n",
      "417\n",
      "418\n",
      "419\n",
      "420\n",
      "421\n",
      "422\n",
      "423\n",
      "424\n",
      "425\n",
      "426\n",
      "427\n",
      "428\n",
      "429\n",
      "430\n",
      "431\n",
      "432\n"
     ]
    }
   ],
   "source": [
    "PEA_output = []\n",
    "\n",
    "for map_idx, curr_map in enumerate(map_list_unique):\n",
    "    print(map_idx)\n",
    "    \n",
    "    for entry in pathway_list:\n",
    "        if curr_map == entry[0]:\n",
    "            path_name = entry[1]\n",
    "            break\n",
    "\n",
    "\n",
    "    is_sig_in_pathway=0\n",
    "    is_sig_not_pathway=0\n",
    "    not_sig_in_pathway=0\n",
    "    not_sig_not_pathway=0\n",
    "\n",
    "    for met_inx, met_data in enumerate(Metabolomics_data):\n",
    "        S1 = met_data['SHAM']\n",
    "        S2 = met_data['RYGB']\n",
    "\n",
    "        ttest_result = sp.ttest_ind(S1,S2)\n",
    "        ttest_result = ttest_result[1]\n",
    "\n",
    "        KEGG_id = met_data['KEGG_id']\n",
    "\n",
    "        in_pathway =0\n",
    "        is_sig=0\n",
    "\n",
    "        for i in range(len(pathway_cpd)):\n",
    "            if pathway_cpd[i][0] == KEGG_id and pathway_cpd[i][1] == curr_map:\n",
    "                in_pathway = 1\n",
    "                break\n",
    "\n",
    "        if ttest_result<0.05:\n",
    "            is_sig = 1\n",
    "\n",
    "        if in_pathway ==1 and is_sig ==1:\n",
    "            is_sig_in_pathway+=1\n",
    "        elif in_pathway==1 and is_sig ==0:\n",
    "            not_sig_in_pathway+=1\n",
    "        elif in_pathway==0 and is_sig ==1:\n",
    "            is_sig_not_pathway+=1\n",
    "        elif in_pathway==0 and is_sig==0:\n",
    "            not_sig_not_pathway+=1\n",
    "\n",
    "    contingency_table = [[is_sig_in_pathway,is_sig_not_pathway],[not_sig_in_pathway,not_sig_not_pathway]]\n",
    "    oddsratio,pvalue = sp.fisher_exact(contingency_table)\n",
    "    \n",
    "    output_list = [curr_map, path_name, pvalue]\n",
    "    PEA_output.append(output_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>KEGG MAP</th>\n",
       "      <th>Pathway Name</th>\n",
       "      <th>pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>map04216</td>\n",
       "      <td>Ferroptosis</td>\n",
       "      <td>0.000220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>map04714</td>\n",
       "      <td>Thermogenesis</td>\n",
       "      <td>0.000253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>map04080</td>\n",
       "      <td>Neuroactive ligand-receptor interaction</td>\n",
       "      <td>0.000746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>map00480</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "      <td>0.003199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>map04725</td>\n",
       "      <td>Cholinergic synapse</td>\n",
       "      <td>0.003626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>map00633</td>\n",
       "      <td>Nitrotoluene degradation</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>map05130</td>\n",
       "      <td>Pathogenic Escherichia coli infection</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>map04071</td>\n",
       "      <td>Sphingolipid signaling pathway</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>map05150</td>\n",
       "      <td>Staphylococcus aureus infection</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>map07110</td>\n",
       "      <td>Benzoic acid family</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     KEGG MAP                             Pathway Name    pvalue\n",
       "0    map04216                              Ferroptosis  0.000220\n",
       "1    map04714                            Thermogenesis  0.000253\n",
       "2    map04080  Neuroactive ligand-receptor interaction  0.000746\n",
       "3    map00480                   Glutathione metabolism  0.003199\n",
       "4    map04725                      Cholinergic synapse  0.003626\n",
       "..        ...                                      ...       ...\n",
       "428  map00633                 Nitrotoluene degradation  1.000000\n",
       "429  map05130    Pathogenic Escherichia coli infection  1.000000\n",
       "430  map04071           Sphingolipid signaling pathway  1.000000\n",
       "431  map05150          Staphylococcus aureus infection  1.000000\n",
       "432  map07110                      Benzoic acid family  1.000000\n",
       "\n",
       "[433 rows x 3 columns]"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PEA_output_df=pd.DataFrame(PEA_output, columns=[\"KEGG MAP\", \"Pathway Name\", \"pvalue\"])\n",
    "PEA_output_df = PEA_output_df.sort_values(by=\"pvalue\",ascending=True).reset_index(drop=True)\n",
    "\n",
    "PEA_output_df.to_csv('PEA_output.csv')\n",
    "PEA_output_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'protein_name': 'Tfrc',\n",
       " 'SHAM': array([6654.748723909054, 5197.19164247816, 205.36, 9677.85875095262,\n",
       "        205.36, 205.36], dtype=object),\n",
       " 'RYGB': array([58995.58730944913, 56665.446395003084, 38887.4445460242,\n",
       "        59931.18380262324, 48842.31475844707, 74630.65750219219],\n",
       "       dtype=object)}"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "swath_lod[0]\n",
    "#hsa_gene_dict\n",
    "#swath_lod"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "hsa_pathway = REST.kegg_link(\"pathway\",\"hsa\").read()\n",
    "hsa_pathway = hsa_pathway.split('\\n')\n",
    "hsa_pathway_list = []\n",
    "for entry in hsa_pathway:\n",
    "    entry_split = entry.split('\\t')\n",
    "    if len(entry_split)==2:\n",
    "        hsa_id = entry_split[0]\n",
    "        path_id = \"map\" + entry_split[1][8:]\n",
    "        curr_list = [hsa_id,path_id]\n",
    "        hsa_pathway_list.append(curr_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "map_list = [entry[1] for entry in hsa_pathway_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "map_list_unique = list(set(map_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'gene_names_vec': ['THRSP', 'LPGP1', 'Lpgp', 'S14', 'SPOT14', 'THRP'],\n",
       " 'gene_description': 'thyroid hormone responsive',\n",
       " 'hsa_id': 'hsa:7069'}"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hsa_gene_dict['LPGP1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "30\n",
      "31\n",
      "32\n",
      "33\n",
      "34\n",
      "35\n",
      "36\n",
      "37\n",
      "38\n",
      "39\n",
      "40\n",
      "41\n",
      "42\n",
      "43\n",
      "44\n",
      "45\n",
      "46\n",
      "47\n",
      "48\n",
      "49\n",
      "50\n",
      "51\n",
      "52\n",
      "53\n",
      "54\n",
      "55\n",
      "56\n",
      "57\n",
      "58\n",
      "59\n",
      "60\n",
      "61\n",
      "62\n",
      "63\n",
      "64\n",
      "65\n",
      "66\n",
      "67\n",
      "68\n",
      "69\n",
      "70\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "81\n",
      "82\n",
      "83\n",
      "84\n",
      "85\n",
      "86\n",
      "87\n",
      "88\n",
      "89\n",
      "90\n",
      "91\n",
      "92\n",
      "93\n",
      "94\n",
      "95\n",
      "96\n",
      "97\n",
      "98\n",
      "99\n",
      "100\n",
      "101\n",
      "102\n",
      "103\n",
      "104\n",
      "105\n",
      "106\n",
      "107\n",
      "108\n",
      "109\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "116\n",
      "117\n",
      "118\n",
      "119\n",
      "120\n",
      "121\n",
      "122\n",
      "123\n",
      "124\n",
      "125\n",
      "126\n",
      "127\n",
      "128\n",
      "129\n",
      "130\n",
      "131\n",
      "132\n",
      "133\n",
      "134\n",
      "135\n",
      "136\n",
      "137\n",
      "138\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "143\n",
      "144\n",
      "145\n",
      "146\n",
      "147\n",
      "148\n",
      "149\n",
      "150\n",
      "151\n",
      "152\n",
      "153\n",
      "154\n",
      "155\n",
      "156\n",
      "157\n",
      "158\n",
      "159\n",
      "160\n",
      "161\n",
      "162\n",
      "163\n",
      "164\n",
      "165\n",
      "166\n",
      "167\n",
      "168\n",
      "169\n",
      "170\n",
      "171\n",
      "172\n",
      "173\n",
      "174\n",
      "175\n",
      "176\n",
      "177\n",
      "178\n",
      "179\n",
      "180\n",
      "181\n",
      "182\n",
      "183\n",
      "184\n",
      "185\n",
      "186\n",
      "187\n",
      "188\n",
      "189\n",
      "190\n",
      "191\n",
      "192\n",
      "193\n",
      "194\n",
      "195\n",
      "196\n",
      "197\n",
      "198\n",
      "199\n",
      "200\n",
      "201\n",
      "202\n",
      "203\n",
      "204\n",
      "205\n",
      "206\n",
      "207\n",
      "208\n",
      "209\n",
      "210\n",
      "211\n",
      "212\n",
      "213\n",
      "214\n",
      "215\n",
      "216\n",
      "217\n",
      "218\n",
      "219\n",
      "220\n",
      "221\n",
      "222\n",
      "223\n",
      "224\n",
      "225\n",
      "226\n",
      "227\n",
      "228\n",
      "229\n",
      "230\n",
      "231\n",
      "232\n",
      "233\n",
      "234\n",
      "235\n",
      "236\n",
      "237\n",
      "238\n",
      "239\n",
      "240\n",
      "241\n",
      "242\n",
      "243\n",
      "244\n",
      "245\n",
      "246\n",
      "247\n",
      "248\n",
      "249\n",
      "250\n",
      "251\n",
      "252\n",
      "253\n",
      "254\n",
      "255\n",
      "256\n",
      "257\n",
      "258\n",
      "259\n",
      "260\n",
      "261\n",
      "262\n",
      "263\n",
      "264\n",
      "265\n",
      "266\n",
      "267\n",
      "268\n",
      "269\n",
      "270\n",
      "271\n",
      "272\n",
      "273\n",
      "274\n",
      "275\n",
      "276\n",
      "277\n",
      "278\n",
      "279\n",
      "280\n",
      "281\n",
      "282\n",
      "283\n",
      "284\n",
      "285\n",
      "286\n",
      "287\n",
      "288\n",
      "289\n",
      "290\n",
      "291\n",
      "292\n",
      "293\n",
      "294\n",
      "295\n",
      "296\n",
      "297\n",
      "298\n",
      "299\n",
      "300\n",
      "301\n",
      "302\n",
      "303\n",
      "304\n",
      "305\n",
      "306\n",
      "307\n",
      "308\n",
      "309\n",
      "310\n",
      "311\n",
      "312\n",
      "313\n",
      "314\n",
      "315\n",
      "316\n",
      "317\n",
      "318\n",
      "319\n",
      "320\n",
      "321\n",
      "322\n",
      "323\n",
      "324\n",
      "325\n",
      "326\n",
      "327\n",
      "328\n",
      "329\n",
      "330\n",
      "331\n",
      "332\n",
      "333\n",
      "334\n",
      "335\n",
      "336\n"
     ]
    }
   ],
   "source": [
    "PEA_output_SWATH = []\n",
    "\n",
    "for map_idx, curr_map in enumerate(map_list_unique):\n",
    "    print(map_idx)\n",
    "    \n",
    "    for entry in pathway_list:\n",
    "        if curr_map == entry[0]:\n",
    "            path_name = entry[1]\n",
    "            break\n",
    "\n",
    "\n",
    "    is_sig_in_pathway=0\n",
    "    is_sig_not_pathway=0\n",
    "    not_sig_in_pathway=0\n",
    "    not_sig_not_pathway=0\n",
    "\n",
    "    for met_inx, prot_data in enumerate(swath_lod):\n",
    "        S1 = prot_data['SHAM']\n",
    "        S2 = prot_data['RYGB']\n",
    "        prot = str(prot_data['protein_name']).upper()\n",
    "        \n",
    "        if prot in hsa_gene_dict:\n",
    "            hsa_id = hsa_gene_dict[prot]['hsa_id']\n",
    "            ttest_result = sp.ttest_ind(S1,S2)\n",
    "            ttest_result = ttest_result[1]\n",
    "\n",
    "\n",
    "\n",
    "            in_pathway =0\n",
    "            is_sig=0\n",
    "\n",
    "            for i in range(len(hsa_pathway_list)):\n",
    "                if hsa_pathway_list[i][0] == hsa_id and hsa_pathway_list[i][1] == curr_map:\n",
    "                    in_pathway = 1\n",
    "                    break\n",
    "\n",
    "            if ttest_result<0.05:\n",
    "                is_sig = 1\n",
    "\n",
    "            if in_pathway ==1 and is_sig ==1:\n",
    "                is_sig_in_pathway+=1\n",
    "            elif in_pathway==1 and is_sig ==0:\n",
    "                not_sig_in_pathway+=1\n",
    "            elif in_pathway==0 and is_sig ==1:\n",
    "                is_sig_not_pathway+=1\n",
    "            elif in_pathway==0 and is_sig==0:\n",
    "                not_sig_not_pathway+=1\n",
    "\n",
    "    contingency_table = [[is_sig_in_pathway,is_sig_not_pathway],[not_sig_in_pathway,not_sig_not_pathway]]\n",
    "    oddsratio,pvalue = sp.fisher_exact(contingency_table)\n",
    "    \n",
    "    output_list = [curr_map, path_name, pvalue]\n",
    "    PEA_output_SWATH.append(output_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "PEA_output_SWATH_df = pd.DataFrame(PEA_output_SWATH, columns = ['KEGG_Map', 'Pathway Name', 'pvalue'])\n",
    "PEA_output_SWATH_df = PEA_output_SWATH_df.sort_values(by=['pvalue'])\n",
    "PEA_output_SWATH_df.to_csv('PEA_SWATH.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7.1 64-bit ('chand': virtualenv)",
   "language": "python",
   "name": "python37164bitchandvirtualenv9dc6a78812824289974a0a4aa2dbb22b"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
